crates/fortitude_linter/src/rules/style/inconsistent_dimension.rs

```diff
@@ -0,0 +1,393 @@
+        .iter()
+        .any(|attr| attr.kind().is_dimension())
+    {
+        return None;
+    }
+
+    Some(
+        decl_line
+            .names()
+            .iter()
+            .filter(|name| name.size().is_some())
+            .map(|node| {
+                let mut edits = fix_bad_array_decl(node, decl_line, src, true).ok().unwrap();
+                Diagnostic::from_node(
+                    BadArrayDeclaration {
+                        prefer_attribute: true,
+                    },
+                    node.node(),
+                )
+                .with_fix(Fix::unsafe_edits(edits.remove(0), edits))
+            })
+            .collect_vec(),
+    )
+}
+
+fn check_bad_array_decl_prefer_no_attribute(
+    decl_line: &VariableDeclaration,
+    src: &SourceFile,
+) -> Option<Vec<Diagnostic>> {
+    // If there's no `dimension` attribute, there's nothing to do
+    decl_line
+        .attributes()
+        .iter()
+        .find(|attr| attr.kind().is_dimension())?;
+
+    Some(
+        decl_line
+            .names()
+            .iter()
+            .filter(|name| name.size().is_none())
+            .map(|node| {
+                let mut edits = fix_bad_array_decl(node, decl_line, src, false)
+                    .ok()
+                    .unwrap();
+                Diagnostic::from_node(
+                    BadArrayDeclaration {
+                        prefer_attribute: false,
+                    },
+                    node.node(),
+                )
+                .with_fix(Fix::unsafe_edits(edits.remove(0), edits))
+            })
+            .collect_vec(),
+    )
+}
+
+pub fn check_inconsistent_dimension_rules(
+    settings: &CheckSettings,
+    decl_line: &VariableDeclaration,
+    src: &SourceFile,
+) -> Vec<Diagnostic> {
+    let mut violations = Vec::new();
+
+    let rules = &settings.rules;
+    let prefer_attribute = settings.inconsistent_dimension.prefer_attribute;
+
+    if rules.enabled(Rule::InconsistentArrayDeclaration) {
+        if let Some(violation) = check_inconsistent_dimension(decl_line, src) {
+            violations.extend(violation);
+        }
+    }
+    if rules.enabled(Rule::MixedScalarArrayDeclaration) {
+        if let Some(violation) = check_mixed_scalar_array(decl_line, src) {
+            violations.extend(violation);
+        }
+    }
+    if rules.enabled(Rule::BadArrayDeclaration) {
+        if prefer_attribute {
+            if let Some(violation) = check_bad_array_decl_prefer_attribute(decl_line, src) {
+                violations.extend(violation);
+            }
+        } else if let Some(violation) = check_bad_array_decl_prefer_no_attribute(decl_line, src) {
+            violations.extend(violation);
+        }
+    }
+    violations
+}
+
+pub mod settings {
+    use crate::display_settings;
+    use ruff_macros::CacheKey;
+    use std::fmt::Display;
+
+    #[derive(Debug, Clone, Default, CacheKey)]
+    pub struct Settings {
+        pub prefer_attribute: bool,
```

Comment:
As we discussed that this rule might be a little too opinionated to appear when the user asks for `style`/`ALL`, would it be possible to instead turn this into a ternary choice of `Some(true)`, `Some(false)`, and `None`, with `None` meaning 'either'? That way, the rule wouldn't do anything unless the user sets this up in their settings file.

If that won't work well with the TOML file, maybe string options of 'always', 'never', 'sometimes'/'either'/'any'?